mllib/src/main/scala/org/apache/spark/mllib/feature/IDF.scala

```diff
@@ -218,7 +218,7 @@ private object IDFModel {
           newValues(k) = values(k) * idf(indices(k))
           k += 1
         }
-        Vectors.sparse(n, indices, newValues)
+        Vectors.sparse(n, indices, newValues).toSparse
```

Comment:
I could imagine updating [this line](https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala#L307) in `Vectors.sparse` to something like:
```scala
val (indices, values) = elements.filter(_._2 != 0).sortBy(_._1).unzip
```